wled00/FX_fcn.cpp

```diff
@@ -760,16 +779,26 @@ void IRAM_ATTR Segment::setPixelColor(int i, uint32_t col)
         indexMir += offset; // offset/phase
         if (indexMir >= stop) indexMir -= len; // wrap
 #ifndef WLED_DISABLE_MODE_BLEND
-        if (_modeBlend) tmpCol = color_blend(strip.getPixelColor(indexMir), col, 0xFFFFU - progress(), true);
+        if (_modeBlend) {
```

Comment:
This is entirely wrong approach.
`_modeBlend` is a flag to indicate that `setPixelColor()` needs to take into account underlying pixel color. If you want to implement correct *transitioning* you need to take into account that segments may overlap.

And you guessed it correctly, this is the place where your *color* transitions should apply.

Reply:
Happy to decouple my implementation from `_modeBlend`. I was building on top of your blending implementation, since what you implemented is basically the default transition style in this PR. A lot of the effect transition code was reused in my implementation.

Reply:
See my other comment regarding `SEGCOLOR()` macro.
With current *mode blending* I managed to do actual effect transitions from one effect to another in a smooth way without any change to effect functions. The goal of this PR should be the same while allowing different transition styles.
And these styles need to apply to color (possibly taken from palette as well) and to effects.

So you effectively have two transitioning methods to implement.